src/FakeItEasy/Core/DefaultExceptionThrower.cs

```diff
@@ -50,21 +50,26 @@ public void ThrowFailedToGenerateProxyWithResolvedConstructors(Type typeOfFake,
                 .AppendLine(".")
                 .AppendLine()
                 .AppendIndented("  ", "Below is a list of reasons for failure per attempted constructor:")
-                .AppendLine()
-                .AppendIndented("    ", "No constructor arguments failed:")
-                .AppendLine()
-                .AppendIndented("      ", reasonForFailureOfUnspecifiedConstructor)
                 .AppendLine();
 
             if (resolvedConstructors.Any(x => x.WasSuccessfullyResolved))
             {
                 foreach (var constructor in resolvedConstructors.Where(x => x.WasSuccessfullyResolved))
                 {
-                    message
-                        .AppendIndented("    ", "Constructor with signature (")
-                        .Append(constructor.Arguments.ToCollectionString(x => x.ArgumentType.ToString(), ", "))
-                        .AppendLine(") failed:")
-                        .AppendIndented("      ", constructor.ReasonForFailure)
+                    if (constructor.Arguments == null)
+                    {
+                        message
+                            .AppendIndented("    ", "No constructor arguments failed:");

```

Comment:
I don't love it either. I'd prefer "Paramterless constructor failed", or similar, or remove the special case and just indicate that it's the one with the signature `()`.

Reply:
However, I'm preserving the existing message. Happy to change, but would rather do in another issue, for visibility.

Reply:
> remove the special case and just indicate that it's the one with the signature `()`.

I like that!

> However, I'm preserving the existing message.

Ah, I didn't realize. Sure, it can be done in a separate PR.

Reply:
#1373